stl/src/cthread.cpp

```diff
@@ -103,9 +105,45 @@ _CRTIMP2_PURE _Thrd_id_t __cdecl _Thrd_id() noexcept { // return unique id for c
 }
 
 _CRTIMP2_PURE unsigned int __cdecl _Thrd_hardware_concurrency() noexcept { // return number of processors
-    SYSTEM_INFO info;
-    GetNativeSystemInfo(&info);
-    return info.dwNumberOfProcessors;
+    // Most devices have only one processor group and thus have the same buffer_size
+#if defined(_M_X64) || defined(_M_ARM64) // 16 bytes per group
+    constexpr int stack_buffer_size = 48;
+#elif defined(_M_IX86) || defined(_M_ARM) // 12 bytes per group
+    constexpr int stack_buffer_size = 44;
+#else
+    constexpr int stack_buffer_size = 0;

```

Comment:
This will attempt to allocate zero-sized array. I wouldn't do that, and `#error` here directly